infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/AbstractWorkspaceServiceAccount.java

```diff
@@ -180,11 +194,17 @@ private void ensureExplicitClusterRoleBindings(Client k8sClient) {
    *
    * @param name the name of the role
    * @param resources the resources the role grants access to
+   * @param resourceNames specific resource names witch the role grants access to. {@code null} if
+   *     empty.
    * @param verbs the verbs the role allows
    * @return the role object for the given type of Client
    */
   protected abstract R buildRole(
-      String name, List<String> resources, List<String> apiGroups, List<String> verbs);
+      String name,
+      List<String> resources,
+      @Nullable List<String> resourceNames,
```

Comment:
what do others think about not have this `@Nullable`, but rather force caller to pass emptyList()?
cc: @skabashnyuk @mshaposhnik @xbaran4 @metlos

Reply:
+1 for empty.

Reply:
Reworked to empty